File: src/main/java/com/zhmyr/parser/Body.java
package com.zhmyr.parser;

import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import java.util.List;
import java.util.Map;

class Body {
    private String title;
    private Image img;
    private List<Section> mSection;
    private Map<String, String> notes;

    public Body(Node item) {
        NodeList child = item.getChildNodes();
        for (int i = 0; i < child.getLength(); i++) {
            switch (child.item(i).getNodeName()) {
                case "image":
                    if (img == null) img = new Image(child.item(i));
                    break;
                case "title":
                    if (title == null) title = child.item(i).getTextContent();
                    break;
                case "epigraph":

            }
        }
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<Section> getSection() {
        return mSection;
    }

    public Map<String, String> getNotes() {
        return notes;
    }

    public boolean addSection(Section section) {
        return this.mSection.add(section);
    }

    public String addNote(String key, String value) {
        return this.notes.put(key, value);
    }
}

File: src/main/java/com/zhmyr/parser/Body.kt
package com.zhmyr.parser

import com.sun.istack.internal.NotNull
import com.sun.istack.internal.Nullable
import org.w3c.dom.Node
import java.util.*

//http://www.fictionbook.org/index.php/Элемент_body
class Body internal constructor(body: Node) {
    @get:Nullable
    var lang: String? = null
    @get:Nullable
    var name: String? = null
    @get:Nullable
    var title: Title? = null
    @get:Nullable
    var image: Image? = null
    @get:NotNull
    var sections =
        listOf<Section>()
    @get:Nullable
    var epigraphs: ArrayList<Epigraph>? = null

    init {
        val section = mutableListOf<Section>()
        val attrs = body.attributes
        for (index in 0 until attrs.length) {
            val attr = attrs.item(index)
            if (attr.nodeName == "name") {
                name = attr.nodeValue
            }
            if (attr.nodeName == "xml:lang") {
                lang = attr.nodeValue
            }
        }
        val map = body.childNodes
        for (index in 0 until map.length) {
            val node = map.item(index)
            when (node.nodeName) {
                "section" -> section += Section(node)
                "img" -> title = Title(node)
                "name" -> name = node.textContent
                "image" -> image = Image(node)
                "epigraph" -> {
                    if (epigraphs == null) epigraphs = ArrayList()
                    epigraphs!!.add(Epigraph(node))
                }
            }
        }
        sections = section
    }
}